api_types.ts

```diff
@@ -667,6 +667,10 @@ export interface FetchRequest {
    * You may inspect the `Location` header of the response to observe the indicated redirect URL.
    */
   ignoreRedirects?: boolean;
+  /**
+   * If true, the response {@link FetchResponse.body} will not be automatically decompressed.
+   */
+  disableDecompression?: boolean;
```

Comment:
This feels kind of weird as an interface. It seems like this only exists for the case where decompressing causes an OOM, in which case, what is the pack going to do with a compressed response that they can't decompress? The maker is just stuck.

I think we should just have a flag that controls whether or not this functionality is enabled (we have lots of precedent for passing flags into the lambda) and turn it on and monitor for new issues, and deal with whatever we encounter. I think the burden will be on us inside the platform to temporarily add per-pack overrides to stamp our OOMs and then fixing or suggesting fixes to use smaller page sizes or whatever.

Reply:
SGTM, removed.

Reply:
SGTM, removed.

Reply:
FWIW I think there is value in allowing developer to disable our automatic compression / decompression feature. Specifically, when a Pack is just trying to shuttle data from A to B and doesn't care about the contents, disabling compression could allow it to stay under the memory limits. A Pack may still want to use automatic compression/decompression on other requests where it does have to parse the response.

Axios, Got, and node-fetch all have this option, usually called `decompress: true`. If enabled (the default) the behavior seems to be:
- Automatically add an `Accept-Encoding` header with values we can decompress.
- Automatically decompress payloads we can decompress
- Remove the `Content-Encoding` header if we decompressed the body

Reply:
I'm fine with essentially just passing through the behavior of `compress` from node-fetch. The issue with this `disableCompression` interface as proposed was that I understood it to still set `Accept-Encoding` headers on the request so _Coda_ gets the benefit of compression during transport between the fetcher and the API, but that it hands the still-compressed value back to the pack to try to make sense of.

Either way, the immediate issue at hand is fixing OOMs with the compression that we apply by default, and adding net-new optional SDK functionality won't help there, so I think we might as well keep those as separate efforts.